src/components/Home.tsx

```diff
@@ -0,0 +1,163 @@
+import { useQuery } from "@tanstack/react-query";
+import { RankDetails } from "../models/state";
+import { fetchTopManga, fetchTopManhwa, fetchTopManhua } from "../services/api";
+
+function Home() {
+  return (
+    <>
+      <h2 className="text-white font-semibold text-xl">Top Manga</h2>
+      <TopManga />
+      <h2 className="text-white font-semibold text-xl">Top Manhwa</h2>
+      <TopManhwa />
+      <h2 className="text-white font-semibold text-xl">Top Manhua</h2>
+      <TopManhua />
+    </>
+  );
+}
+
+export default Home;
+
+function TopManga() {
+  const { data: topMangaResults, error } = useQuery({
+    queryKey: [`fetchTopManga`],
+    queryFn: () => fetchTopManga(),
+  });
+
+  if (error) {
+    console.error("An unexpected error occurred: " + error);
+  }
+
+  if (!topMangaResults) {
```

Comment:
You should preferably check all the subtypes you use as well so this will become:

```
if (!(
topMangaResults?.[0]?.md_covers?.[0]?.b2key || 
topMangaResults?.[0]?.slug || 
topMangaResults?.[0]?.title
)) 
{...}
```

Same goes for the bottom guard clauses.